Mm metrics

Borg77777 opened the pull request:
Is this close to what you had in mind?

Borg77777 committed:
keep track of market maker metrics

Borg77777 committed:
Merge branch 'master' into mm-metrics

# Conflicts:
#	model.py
#	objects/trader.py
#	simulate_model.py

joerischasfoort commented:
Welll, in spirit yes. In execution, almost. While, the structure that you propose here seems to work. It is more complicated than what I use for the traders. Also, if you do it the same way as I did with the traders there is more consistency in the code. Could you please do that? Check out my TODO that I placed on line 34. Just place it there  with market.maker.var.money.append( etc.... Then on line 58 you can just use the latest addition market_maker.var.stocks[-1] (instead of 0).

joerischasfoort commented:
Also, I cannot find the graphs that track the market maker statistics. Could you add them to the notebook? Or are you not comfortable with iPython Notebooks? The reason why I like this approach is that I can directly see the results and don't have to run the model for myself.

Borg77777 committed:
structure revisions

Borg77777 committed:
added market maker class and RL agent

joerischasfoort committed:
noise_for_rl_model

- made some syntax changes
- removed ticks per turn
- simplified parameter values
- created new version of model where noise traders can only post int price orders
- added money multiplier parameter to stabilize price
- modified noise trader only notebook.
